crates/astria-sequencer/src/grpc/sequencer.rs

```diff
@@ -118,14 +118,21 @@ impl SequencerService for SequencerServer {
                 ))
             })?;
 
-        let all_rollup_ids = snapshot
+        let mut all_rollup_ids = snapshot
             .get_rollup_ids_by_block_hash(&block_hash)
             .await
             .map_err(|e| Status::internal(format!("failed to get rollup ids from storage: {e}")))?
             .into_iter()
             .map(RollupId::to_vec)
```

Comment:
IMO this should be done at the end of the method, when the final return value is constructed. You can avoid the extra allocation when looking up the IDs, and it's neater to read.